sui/its.js

```diff
@@ -0,0 +1,220 @@
+            tx.pure.address(channelId),
+            tx.pure(bcs.vector(bcs.u8()).serialize(arrayify(payload)).toBytes()),
+        ],
+    });
+
+    const { module_name: moduleName, name, package_id: packageId } = txFields.function.fields;
+
+    // Build the arguments for the move call
+    // There're 5 types of arguments as mentioned in the following link https://github.com/axelarnetwork/axelar-cgp-sui/blob/72579e5c7735da61d215bd712627edad562cb82a/src/bcs.ts#L44-L49
+    const txArgs = txFields.arguments.map(([argType, ...arg]) => {
+        if (argType === 0) {
+            return tx.object(Buffer.from(arg).toString('hex'));
+        } else if (argType === 1) {
+            // TODO: handle pures followed by the bcs encoded form of the pure
+            throw new Error('Not implemented yet');
+        } else if (argType === 2) {
+            return approvedMessage;
+        } else if (argType === 3) {
+            // TODO: handle the payload of the contract call (to be passed into the intermediate function)
+            throw new Error('Not implemented yet');
+        } else if (argType === 4) {
+            // TODO: handle an argument returned from a previous move call, followed by a u8 specified which call to get the return of (0 for the first transaction AFTER the one that gets ApprovedMessage out), and then another u8 specifying which argument to input.
+            throw new Error('Not implemented yet');
+        }
+
+        throw new Error(`Unknown argument type: ${argType}`);
+    });
+
+    // Execute the move call dynamically based on the transaction object
+    tx.moveCall({
+        target: `${packageId}::${moduleName}::${name}`,
+        arguments: txArgs,
+    });
+
+    await broadcast(client, keypair, tx, 'Call Executed');
+}
+
+async function sendTokenDeployment(keypair, client, contracts, args, options) {}
+
+async function receiveTokenDeployment(keypair, client, contracts, args, options) {}
+
+async function mintToken(keypair, client, contracts, args, options) {}
+
+async function processCommand(command, chain, args, options) {
+    const [keypair, client] = getWallet(chain, options);
+
+    await printWalletInfo(keypair, client, chain, options);
+
+    const contracts = [chain.contracts.Example, chain.contracts.GasService, chain.contracts.AxelarGateway];
+
+    await command(keypair, client, contracts, args, options);
+}
+
+async function mainProcessor(command, options, args, processor) {
+    const config = loadConfig(options.env);
+    await processor(command, config.sui, args, options);
+    saveConfig(config, options.env);
+}
+
+if (require.main === module) {
+    const program = new Command();
+    program.name('GMP').description('Example of SUI ITS commands');
+
+    const sendTokenTransferProgram = new Command()
+        .name('sendTokenTransfer')
+        .description('Send token transfer')
+        .command('sendTokenTransfer <destChain> <destContractAddress> <feeAmount> <payload>')
```

Comment:
```suggestion
        .command('interchain-transfer <destination-chain> <destination-address> <fee> <payload>')
```
let's use kebab casing for cli args going forward